### PR TITLE
[PolicyInsights] Add tests

### PR DESCRIPTION
**Pull request by dvbb**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/policyinsights/Azure.ResourceManager.PolicyInsights/tests/Scenario/PolicyMetadataTests.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        }
+
+        [SetUp]
+        public void TestSetUp()
+        {
+            _metadataCollection = DefaultTenant.GetAllPolicyMetadata();
+        }
+
+        [RecordedTest]
+        [Ignore("Deserialize Error: System.UriFormatException: Invalid URI: The URI is empty.")]
```
Comment (dvbb) [inline sdk/policyinsights/Azure.ResourceManager.PolicyInsights/tests/Scenario/PolicyMetadataTests.cs:29]:
```json
{
  "properties": {
    ......
    "additionalContentUrl": "7"
  },
  "id": "/providers/Microsoft.PolicyInsights/policyMetadata/NZISM_Security_Benchmark_v1.1_SS-2",
  "name": "NZISM_Security_Benchmark_v1.1_SS-2",
  "type": "Microsoft.PolicyInsights/policyMetadata"
}
```
We have a rename rule in autorest.md that changes all '*Uri' types to Uri. There has a property `additionalContentUrl` in PolicyMetadata, and lots of its values will return string.empty, and a special item showed above, will return "7" therefore GetAll() method failed. How should we fix this

File: sdk/policyinsights/Azure.ResourceManager.PolicyInsights/tests/Scenario/PolicyAttestationTests.cs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            Assert.IsNotEmpty(list);
+            ValidateAttestation(list.FirstOrDefault().Data, attestationName);
+
+            // Delete
+            await attestation.DeleteAsync(WaitUntil.Completed);
+            flag = await policyAttestationCollection.ExistsAsync(attestationName);
+            Assert.IsFalse(flag);
+        }
+
+        private void ValidateAttestation(PolicyAttestationData attestation, string attestationName)
```
Comment (ArthurMa1978) [inline sdk/policyinsights/Azure.ResourceManager.PolicyInsights/tests/Scenario/PolicyAttestationTests.cs:131]: not check the id of the `PolicyAttestationData` ?

File: sdk/policyinsights/Azure.ResourceManager.PolicyInsights/tests/Scenario/PolicyRemediationTests.cs [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.PolicyInsights.Tests
+{
+    internal class PolicyRemediationTests : PolicyInsightsManagementTestBase
+    {
+        private const string _remediationPrefixName = "remediation";
+        private PolicyAssignmentResource _policyAssignment;
+
+        public PolicyRemediationTests(bool isAsync) : base(isAsync)//, RecordedTestMode.Record)
```
Comment (ArthurMa1978) [inline sdk/policyinsights/Azure.ResourceManager.PolicyInsights/tests/Scenario/PolicyRemediationTests.cs:22]: please remove the useless code

File: sdk/policyinsights/Azure.ResourceManager.PolicyInsights/tests/Scenario/PolicyMetadataTests.cs (hunk 1 above)
Comment (ArthurMa1978) [reply sdk/policyinsights/Azure.ResourceManager.PolicyInsights/tests/Scenario/PolicyMetadataTests.cs:29]: If the value of this property is not always a real `Uri`, we should use custom code to add a string value property like `additionalContentUriString` and EB Nerver the existing `additionalContentUri`

**Review by archerzz — APPROVED** [review]

File: sdk/policyinsights/Azure.ResourceManager.PolicyInsights/tests/Scenario/PolicyMetadataTests.cs (hunk 1 above)
Comment (dvbb) [reply sdk/policyinsights/Azure.ResourceManager.PolicyInsights/tests/Scenario/PolicyMetadataTests.cs]: Resolved in [this commit](https://github.com/Azure/azure-sdk-for-net/pull/34797/commits/893d0f57cbcd01edb7afef4ad244eddf95976d5d), please review thanks

File: sdk/policyinsights/Azure.ResourceManager.PolicyInsights/src/Customized/Models/SlimPolicyMetadata.cs [hunk 4]
```diff
@@ -49,7 +50,12 @@ internal SlimPolicyMetadata(ResourceIdentifier id, string name, ResourceType res
         /// <summary> The owner of the policy metadata. </summary>
         public string Owner { get; }
         /// <summary> Url for getting additional content about the resource metadata. </summary>
-        public Uri AdditionalContentUri { get; }
+        [EditorBrowsable(EditorBrowsableState.Never)]
```
Comment (ArthurMa1978) [inline sdk/policyinsights/Azure.ResourceManager.PolicyInsights/src/Customized/Models/SlimPolicyMetadata.cs:53]: Add `Obsolete` here

File: sdk/policyinsights/Azure.ResourceManager.PolicyInsights/src/Customized/Models/SlimPolicyMetadata.cs [hunk 5]
```diff
@@ -49,7 +50,12 @@ internal SlimPolicyMetadata(ResourceIdentifier id, string name, ResourceType res
...
+        public Uri AdditionalContentUri => Uri.IsWellFormedUriString(AdditionalContentUriString, UriKind.Absolute) ? new Uri(AdditionalContentUriString) : null;
```
Comment (ArthurMa1978) [inline sdk/policyinsights/Azure.ResourceManager.PolicyInsights/src/Customized/Models/SlimPolicyMetadata.cs:54]: This is readonly property, so just parse it in the constructor, not here.

File: sdk/policyinsights/Azure.ResourceManager.PolicyInsights/src/Customized/PolicyMetadataData.cs [hunk 6]
```diff
@@ -56,7 +57,12 @@ internal PolicyMetadataData(ResourceIdentifier id, string name, ResourceType res
         /// <summary> The owner of the policy metadata. </summary>
         public string Owner { get; }
         /// <summary> Url for getting additional content about the resource metadata. </summary>
-        public Uri AdditionalContentUri { get; }
+        [EditorBrowsable(EditorBrowsableState.Never)]
```
Comment (ArthurMa1978) [inline sdk/policyinsights/Azure.ResourceManager.PolicyInsights/src/Customized/PolicyMetadataData.cs:60]: Add `Obsolete` here

File: sdk/policyinsights/Azure.ResourceManager.PolicyInsights/src/Customized/PolicyMetadataData.cs [hunk 7]
```diff
@@ -56,7 +57,12 @@ internal PolicyMetadataData(ResourceIdentifier id, string name, ResourceType res
...
+        public Uri AdditionalContentUri => Uri.IsWellFormedUriString(AdditionalContentUriString, UriKind.Absolute) ? new Uri(AdditionalContentUriString) : null;
```
Comment (ArthurMa1978) [inline sdk/policyinsights/Azure.ResourceManager.PolicyInsights/src/Customized/PolicyMetadataData.cs:61]: This is readonly property, so just parse it in the constructor, not here.

File: sdk/policyinsights/Azure.ResourceManager.PolicyInsights/src/Customized/Models/SlimPolicyMetadata.cs (hunk 5 above)
Comment (dvbb) [reply sdk/policyinsights/Azure.ResourceManager.PolicyInsights/src/Customized/Models/SlimPolicyMetadata.cs:54]: image
```
        [Obsolete("This property has been replaced by AdditionalContentUriString", true)]
        public Uri AdditionalContentUri { get; }
```
if we add the [Obsolete] attribute, it cannot be used otherwhere

**Comment by samarthlakhotia** [comment]
Hi @dvbb, I am from the policy insights team and I wanted to ask if there is any ongoing work (or any planned) to migrate other tests in policy insights too?

**Comment by dvbb** [comment]
Hi @samarthlakhotia , I think currently there are no ongoing work or plans for this

**Pull request merged**